README.md

```diff
@@ -1,66 +1,292 @@
 
-## Extract Modules
+## Instructions 
 
-### Memorable
+### A Note on Refactoring Practices
 
-Build a `Memorable` module in `lib/concerns/memorable.rb` that consolidates the logic that allows a class to remember instances of itself in memory.
+We use TDD (test-driven development) for a reason. We write tests to define the desired behavior of our program so that we can write clean, beautiful code. Such code usually *isn't* the code you write the first time around. The code you first right is the code that makes your program *work*, the code that gets those tests passing. Then, we refactor our code to make it clean, DRY, and easy to understand. This is where our tests come in. If we write thorough tests that cover all of the aspects of our code's desired behavior. We can *first* write code that passes those tests and *then* break our code, fail our tests, write better code and pass our tests again. 
 
-You will need to extract the following methods into that module, within a `ClassMethods` sub-module.
+This is called the **red, green, refactor** pattern. First tests fail, then you write bad code to get them to pass, *then* you refactor that bad code into good code. In this lab, you'll start by running the test suite. You'll see that all of the tests pass. Then, we'll break that code in order to refactor it, write better code and get our tests passing again. Remember, don't be afraid of broken code! Broken code is the status quo in programming. Your job is often to break something to make it better. Embrace broken code. 
 
-- `.reset_all`
-- `.count`
+### Step 1: Class Methods
+
+First, run the test suite. Wow, we're passing all of our tests! Okay, not let go of those passing tests because we are about to break our code. 
+
+The first area of refactoring we'll be attacking are the class methods. Notice that both the `Song` and `Artist` class have `.count`, `reset_all` and `find_by_name` class methods. Instead of repeating the same exact code in both classes, let's extract these class methods into a module that we can *extend* into the classes.
+
+Ready to break your code? Comment out the `reset_all`, `count` and `find_by_name` methods in the `Song` and `Artist` class. Run your test suite. Phew! Okay, we did it. That wasn't so bad, was it?
+
+#### The `Memorable` Module
+
+Let's define our module. Create a `concerns` folder inside `lib`. This is where we'll store our modules. Inside the `concerns` folder, create a file called `memorable.rb`. Open up that file and define a module: 
```

Comment:
Why are you putting this stuff in the `concerns` folder? Do students know what that folder is?